js/src/components/tree-select-control/options.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * External dependencies
+ */
+import { CheckboxControl, Flex } from '@wordpress/components';
+import { useState } from '@wordpress/element';
+import { Icon, chevronUp, chevronDown } from '@wordpress/icons';
+import classnames from 'classnames';
+
+/**
+ * This component renders a list of options and its children recursively
+ *
+ * @param {Object} params Component parameters
+ * @param {Option[]} params.options List of options to be rendered
+ * @param {string[]} params.value List of selected values
+ * @param {string} params.parent The parent of these set of options
+ * @param {Function} params.onChange Callback when an option changes
+ */
+const Options = ( {
+	options = [],
+	value = [],
+	parent = '',
+	onChange = () => {},
+} ) => {
+	const [ expanded, setExpanded ] = useState( [] );
+	/**
+	 * Returns true if all the children for the parent are selected
+	 *
+	 * @param {Option} option The parent option to check
+	 */
+	const isParentSelected = ( option ) => {
+		if ( ! option.children ) {
+			return false;
+		}
+
+		return option.children.every(
+			( child ) =>
+				value.includes( child.value ) || isParentSelected( child )
+		);
+	};
+
+	const toggleExpanded = ( option ) => {
+		setExpanded(
+			expanded.includes( option )
+				? expanded.filter( ( el ) => option !== el )
+				: [ ...expanded, option ]
+		);
+	};
+
+	const isRoot = ( option ) => option.value === '';
+
+	return options.map( ( option ) => {
+		const isExpanded =
+			isRoot( option ) || expanded.includes( option.value );
+
+		const hasChildren = !! option.children?.length;
+
+		return (
+			<div
+				key={ `${ parent }-${ option.value }` }
```

Comment:
Actually, we do "support" the duplicated values in the Tree. It will just select all the elements with the same value. 

This could be useful if 2 options are in different groups. Example: Groups will be meals types like Vegan, vegetarian... etc, and a meal could be in two different groups.

I will update the docs

I don't think anyone wants to add duplicated options in the same group tho...  🤔  so the only thing I could add in that case is the option `idx`, but as far as I know, that should be the last resource. 

WDYT?

Reply:
> I don't think anyone wants to add duplicated options in the same group tho... 🤔 so the only thing I could add in that case is the option `idx`, but as far as I know, that should be the last resource.
> WDYT?

The array index is not ideal as React element's `key` prop when the array is given from outside of the component.

In terms of better fault tolerance and flexible design, I think we can add the following `key` param to the `Option` type, and value is also used as the fallback of `key`.

```
@property {string} [key] The unique key for the option. If not specified, it will fall back to using `value`.
```